build-farm/make-adopt-build-farm.sh

```diff
@@ -18,16 +18,40 @@ set -e
 
 PLATFORM_SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-## Sanity check as this script requires various environment variables so fail
-## fast if one or more are not present. MAYBE autodetect defaults in future?
+## autodetect defaults to improve usability when running this for debugging/testing
+
+if [ -z "$ARCHITECTURE"  ]; then
+   ARCHITECTURE=`uname -p`
+   if [       "$OSTYPE" = "cygwin"  ]; then ARCHITECTURE=`uname -m`; fi
+   if [ "$ARCHITECTURE" = "x86_64"  ]; then ARCHITECTURE=x64;        fi
+   if [ "$ARCHITECTURE" = "powerpc" ]; then ARCHITECTURE=ppc64;      fi
+   echo ARCHITECTURE not defined - assuming $ARCHITECTURE
+   export ARCHITECTURE
+fi
+
+if [ -z "$TARGET_OS" ]; then
```

Comment:
Some comments on these if blocks (or extracting them to functions if possible) will help

Reply:
I thought they were self-explanatory but I can add a comment sayting "translate uname output to the names which adoptopenjdk uses"

Reply:
Added commands (and other special cases for Solaris which I didn't include)